src/UKHO.ERPFacade.API/Filters/LogCorrelationIdMiddleware.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using Azure.Core;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace UKHO.ERPFacade.API.Filters
+{
+    [ExcludeFromCodeCoverage]
+    public class LogCorrelationIdMiddleware
+    {
+        public const string XCorrelationIdHeaderKey = "_X-Correlation-ID";
+        private readonly RequestDelegate _next;
+
+        public LogCorrelationIdMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext httpContext)
+        {
+            string? correlationId = string.Empty;
+
+            if (httpContext.Request.ContentLength > 0)
+            {
+                httpContext.Request.EnableBuffering();
+                var bodyAsText = await new StreamReader(httpContext.Request.Body).ReadToEndAsync();
+                JObject json = JObject.Parse(bodyAsText);
+                correlationId = json.SelectToken("data.traceId")?.Value<string>();
```

Comment:
Token id should be a constant

Reply:
LogCorrelationIdMiddleware.cs is no longer in use. functionality from this is merged into CorrelationIdMiddleware.cs.